web-client/src/pages/signup.js

```diff
@@ -11,40 +50,40 @@ export default function Signup(){
+            <label htmlFor="sign-up-form-name">Name</label>
+            <input type="text" id="sign-up-form-name" name="name"  value={name} onChange={e => onChange(e)} required/>
           </div>
           <div className="form-group">
-            <label for="sign-up-form-email">Email</label>
-            <input type="email" name="email" required id="sign-up-form-email"/>
+            <label htmlFor="sign-up-form-email">Email</label>
+            <input type="email" name="email" required id="sign-up-form-email" value={email}  onChange={e => onChange(e)}/>
           </div>
           <div className="form-group">
-            <label for="sign-up-form-password">Password</label>
-            <input type="password" name="password" id="sign-up-form-password"/>
+            <label htmlFor="sign-up-form-password">Password</label>
+            <input type="password" name="password" id="sign-up-form-password" value={password}  onChange={e => onChange(e)}/>
           </div>
           <div className="form-group">
-            <label for="sign-up-form-dob">Date of Birth</label>
-            <input type="date" name="dateOfBirth" id="sign-up-form-dob"/>
+            <label htmlFor="sign-up-form-dob">Date of Birth</label>
+            <input type="date" name="dateOfBirth" id="sign-up-form-dob" value={dateOfBirth}  onChange={e => onChange(e)}/>
           </div>
           <div className="form-group">
             <label>Notification Preference</label>
-            <label><input type="radio" name="notificationPreference"/> Email</label>
-            <label><input type="radio" name="notificationPreference"/> Phone</label>
-            <label><input type="radio" name="notificationPreference"/> None</label>
+            <label><input type="radio" name="notificationPreference" value='email' onChange={onChange} checked={notificationPreference==='email'}/> Email</label>
+            <label><input type="radio" name="notificationPreference" value='phone' onChange={onChange} checked={notificationPreference==='phone'}/> Phone</label>
+            <label><input type="radio" name="notificationPreference" value='none' onChange={onChange} checked={notificationPreference==='none'}/> None</label>
           </div>
-          
+
           <div className="form-group">
-            <label for="sign-up-form-bio">Write About Yourself</label>
-            <textarea name="biography" id="sign-up-form-bio" rows="3"></textarea>
+            <label htmlFor="sign-up-form-bio">Write About Yourself</label>
+            <textarea name="biography" id="sign-up-form-bio" rows="3" value={biography} onChange={e => onChange(e)}></textarea>
           </div>
           <div className="form-group">
             <div>
-              <label><input type="checkbox" name="agreeToTerms" value="true"/> I agree to the terms and conditions</label>
+              <label><input type="checkbox" name="agreeToTerms" value='true' onChange={e =>onChange(e)} /> I agree to the terms and conditions</label>
```

Comment:
Did you see if the checkbox actually checks if you click? I think here you gotta look at checked instead of value.

checked will be either true or false.

```suggestion
              <label><input type="checkbox" name="agreeToTerms" value='true' onChange={e => setInputs({ ...inputs, [event.target.name]: event.target.checked })} checked={agreeToTerms} /> I agree to the terms and conditions</label>
```

Reply:
![image](https://github.com/nbktechworld/full-stack-web-dev/assets/13649002/3a8da28c-3c0d-4568-a488-74921be1c4bb)